catalog/services_state.go

```diff
@@ -320,12 +325,28 @@ func (state *ServicesState) AddServiceEntry(newSvc service.Service) {
 	}
 }
 
+// Returns a service for a given ID if it happens to exist on the current
+// host. Returns an error otherwise.
+func (state *ServicesState) GetServiceByID(id string) (service.Service, error) {
```

Comment:
Probably should be `GetLocalServiceByID()` to make it clear that it's only on this host.